enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/LocalDatabase.java

```diff
@@ -193,4 +213,32 @@ public TransactionCommitProcess getCommitProcess()
     {
         return localCommit;
     }
+
+    private synchronized void stopWithRequirement( AvailabilityRequirement requirement ) throws Throwable
+    {
+        log.info( "Stopping, reason: " + requirement.description() );
+        databaseHealth = null;
+        localCommit = null;
+        dataSourceManager.stop();
+
+        raiseAvailabilityGuard( requirement );
```

Comment:
I am also concerned that a transaction might still be in progress running, just about to start of already blocking on something before this was raised, but either way this should take care of a a few cases. Perhaps the intention is for internal components to continuously poll on the availability guard for all blocking matters and thus eventually get released. Let's revisit in the future if necessary and perhaps add that if missing somewhere.